modmesh/gui/naca.py

```diff
@@ -29,125 +29,24 @@
-    yt = 5.0 * thick * (a0 * xh + a1 * xc + a2 * x2 + a3 * x3 + a4 * x4)
-
-    if pos == 0:
-        xu = xc
-        yu = yt
-        xl = xc
-        yl = -yt
-    else:
-        xc1 = xc[xc <= pos]
-        yc1 = camber / np.power(pos, 2) * xc1 * (2 * pos - xc1)
-        dyc1_dx = camber / np.power(pos, 2) * (2 * pos - 2 * xc1)
-
-        xc2 = xc[xc > pos]
-        yc2 = camber / np.power(1 - pos, 2) * (1 - 2 * pos + xc2) * (1 - xc2)
-        dyc2_dx = camber / np.power(1 - pos, 2) * (2 * pos - 2 * xc2)
-
-        yc = np.concatenate([yc1, yc2])
-        dyc_dx = np.concatenate([dyc1_dx, dyc2_dx])
-        theta = np.arctan(dyc_dx)
-
-        xu = xc - yt * np.sin(theta)
-        yu = yc + yt * np.cos(theta)
-        xl = xc + yt * np.sin(theta)
-        yl = yc - yt * np.cos(theta)
-
-    xr = np.concatenate([xu[::-1], xl[1:]])
-    yr = np.concatenate([yu[::-1], yl[1:]])
-    ret = np.vstack([xr, yr]).T.copy()
-    return ret
-
-
-def calc_sample_points(points):
-    # determine the number of sample points for a cubic bezier curve
-    spacing = 0.01
-    segments = np.hypot((points[:-1, 0] - points[1:, 0]),
-                        (points[:-1, 1] - points[1:, 1])) // spacing
-    nsample = np.where(segments > 2, segments - 1, 2).astype(int)
-    return nsample
-
-
-def linear_approximate(world, points):
-    for it in range(points.shape[0] - 1):
-        world.add_edge(points[it, 0], points[it, 1], 0,
-                       points[it + 1, 0], points[it + 1, 1], 0)
-    return
-
-
-def cubic_bezier_approximate(world, points, nsample=None):
-    Vector = core.Vector3dFp64
-
-    if nsample is None:
-        nsample = calc_sample_points(points)
-    for it in range(points.shape[0] - 1):
-        p1 = points[it] + (1 / 3) * (points[it + 1] - points[it])
-        p2 = points[it] + (2 / 3) * (points[it + 1] - points[it])
-
-        b = world.add_bezier([Vector(points[it, 0], points[it, 1], 0),
-                              Vector(p1[0], p1[1], 0),
-                              Vector(p2[0], p2[1], 0),
-                              Vector(points[it + 1, 0], points[it + 1, 1], 0)])
-        b.sample(nsample[it])
-    return
-
-
-def draw_naca4(world, number, npoint, fac, off_x, off_y,
-               use_bezier=False, **kw):
-    crds = calc_naca4_points(number=number, npoint=npoint, **kw)
-    crds *= fac  # scaling factor
-    crds[:, 0] += off_x  # offset in x
-    crds[:, 1] += off_y  # offset in y
-
-    if use_bezier:
-        cubic_bezier_approximate(world, crds)
-    else:
-        linear_approximate(world, crds)
-    return crds
+from modmesh.pilot import airfoil
 
 
 def runmain():
     """
     A simple example for drawing a couple of cubic Bezier curves.
     """
     w = core.WorldFp64()
-    draw_naca4(w, number='0012', npoint=101, fac=5.0, off_x=0.0, off_y=2.0,
-               open_trailing_edge=False, use_bezier=True, cosine_spacing=True)
+    naca4 = airfoil.Naca4(number='0012', open_trailing_edge=False,
```

Comment:
After the PR the code in `modmesh/gui/naca.py` is only responsible for sending arguments to the calculation and drawing helper classes.  We can add GUI controls in this file.

The code implementing calculation (sampling, fitting, etc.) and visualization can be kept in `modmesh/pilot/airfoil/__init__.py`.